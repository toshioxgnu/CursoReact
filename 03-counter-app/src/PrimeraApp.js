import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {

    return(  
        <>
            <pre>{ saludo }</pre>
            <p> { subtitulo } </p>
        </>
    );

}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo : 'Soy un subtitulo'
}

export default PrimeraApp;