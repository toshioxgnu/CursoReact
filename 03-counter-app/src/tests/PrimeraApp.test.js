import React from 'react';
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';


describe('Pruebas en <PrimeraApp />', () => {

    //test('debe mostrar el mensaje Hola Soy Goku', () => {

        //const saludo = 'Hola, soy Goku';

        //const { getByText } = render( <PrimeraApp saludo={ saludo } /> );

        //expect( getByText(saludo) ).toBeInTheDocument();

    //});

    test('Debe de mostrar <PrimeraApp /> Correctamente', () => {
        const saludo = 'Hola, Soy goku';
        const wrapper = shallow( <PrimeraApp saludo={ saludo } /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar el subtitulo enviado por props ', () => {
        const saludo = 'Hola, Soy goku';
        const subtitulo = 'Test Subtitulo';
        const wrapper = shallow( <PrimeraApp 
            saludo={ saludo } 
            subtitulo = {subtitulo}
            /> 
            );
        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subtitulo);
    })
    
    
    
});