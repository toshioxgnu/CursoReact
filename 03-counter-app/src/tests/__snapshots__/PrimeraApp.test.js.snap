// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar <PrimeraApp /> Correctamente 1`] = `
<Fragment>
  <pre>
    Hola, Soy goku
  </pre>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
